File: StructuralPattern/Proxy/Python/Concretes/RealImage.py
from Interfaces import Image


class RealImage(Image):
    """
        Class RealImage implementation for interface image
    """
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fileName = kwargs.get('fileName', None)
        # TODO ...
        if args:
            self.fileName = str(args[0])
        

    def RealImage(self, fileName):
        self.fileName = fileName
        self.selfloadFromDisk()

    def loadFromDisk(self):
        print("Loading " + self.fileName)
    
    def display(self):
        print("Displaying " + self.fileName)

File: StructuralPattern/Proxy/Golang/Concretes/RealImage.go

package concretes
import (
	"fmt"
	"v/Interfaces"
)

/**
 * Class RealImage implementation for interface image
 */
type RealImage struct {
	fileName string
}

// Constructor
func NewRealImage(fileName string) interfaces.Image {
	rI := &RealImage{fileName: fileName}
	rI.LoadFromDisk()
	return rI
}

// Function LoadFromDisk 
func (realImage *RealImage) LoadFromDisk() {
	fmt.Println("Loading " + realImage.fileName)
}

// Deloy method for Image
func (realImage *RealImage) Display() {
	fmt.Println("Displaying " + realImage.fileName)
}
